### ISSUE TITLE
Allow 2D inputs for Qinco

### ISSUE THREAD

**Issue by guillaumeguy** [lucidrains/vector-quantize-pytorch#203]
When the user turns the `implicit_neural_codebook` on, the model ResidualVQ model now expects a 3D tensor inputs. Allowing 2D (B,dim) would be nice.

```python
import torch
from vector_quantize_pytorch import ResidualVQ

residual_vq = ResidualVQ(
    dim = 32,
    num_quantizers = 8,
    codebook_size = 128,
    implicit_neural_codebook = True, # Turning this on make the script fail
    use_cosine_sim = False,
    shared_codebook = False,
)

B = 16
dim = 32
x = torch.randn(B,  dim)

residual_vq.train()

quantized, indices, commit_loss = residual_vq(x, freeze_codebook = False)
quantized_out = residual_vq.get_output_from_indices(indices)
```

Thanks again for all the great work

**Comment by lucidrains** [comment]
@guillaumeguy no problem, want to try [1.22.3](https://github.com/lucidrains/vector-quantize-pytorch/commit/2ebb9d1017e6386a3c045c2e22edfe5d31363084)?

**Comment by guillaumeguy** [comment]
OK, it took me a crazy 2 hours on a A10G to encode 1M vectors with their checkpoint. It's a deal-breaking. I think this implementation is not ready for prime time (plus, it has a restrictive license)
